### PR TITLE
Update dependencies in java function sample

### PR DESCRIPTION
**Pull request by menehune23**
Addresses CVEs discovered in grype scans

### CODE REVIEW

File: java-function/cloudevents/pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
 		<java.version>11</java.version>
```
Comment (trisberg) [inline java-function/cloudevents/pom.xml:11]: shouldn't this be 17?

**Comment by bbtong** [comment]
I think bumping 2020->2022 should also bump 11 -> 17?

File: java-function/cloudevents/pom.xml (hunk 1 above)
Comment (menehune23) [reply java-function/cloudevents/pom.xml]: Fixed

**Comment by menehune23** [comment]
New scan results:

_CloudEvents + Maven_:

```bash
> mvn package && grype target/java-function-0.0.1-full.jar

 ✔ Vulnerability DB        [no update available]
 ✔ Indexed target/java-function-0.0.1-full.jar 
 ✔ Cataloged packages      [7 packages]
 ✔ Scanning image...       [0 vulnerabilities]
   ├── 0 critical, 0 high, 0 medium, 0 low, 0 negligible
   └── 0 fixed

No vulnerabilities found
```

_CloudEvents + Gradle_:

> mvn package && grype target/java-function-0.0.1-full.jar

 ✔ Vulnerability DB        [no update available]
 ✔ Indexed build/libs/java-function-0.0.1.jar 
 ✔ Cataloged packages      [7 packages]
 ✔ Scanning image...       [0 vulnerabilities]
   ├── 0 critical, 0 high, 0 medium, 0 low, 0 negligible
   └── 0 fixed
No vulnerabilities found

```

**Review by trisberg — APPROVED** [review]

**Pull request merged**